servlet-engine/src/main/java/io/micronaut/servlet/engine/DefaultServletHttpRequest.java

```diff
@@ -637,4 +635,62 @@ public <T> Optional<T> get(CharSequence name, ArgumentConversionContext<T> conve
             return Optional.empty();
         }
     }
+
+    /**
+     * Temporary streaming {@link ByteBody} implementation that only supports buffering, for filter
+     * body binding to work. Will be replaced by a proper streaming implementation.
+     */
+    private class MockStreamingBody implements CloseableByteBody {
+        private final AtomicReference<CompletableFuture<? extends CloseableAvailableByteBody>> buffered = new AtomicReference<>();
+
+        @Override
+        public void close() {
+        }
+
+        @Override
+        public @NonNull CloseableByteBody split(SplitBackpressureMode backpressureMode) {
+            return this;
+        }
+
+        @Override
+        public @NonNull OptionalLong expectedLength() {
+            return OptionalLong.empty();
+        }
+
+        @Override
+        public @NonNull InputStream toInputStream() {
+            throw new UnsupportedOperationException("Streaming access not yet implemented for servlet");
+        }
+
+        @Override
+        public @NonNull Publisher<byte[]> toByteArrayPublisher() {
+            throw new UnsupportedOperationException("Streaming access not yet implemented for servlet");
+        }
+
+        @Override
+        public @NonNull Publisher<ByteBuffer<?>> toByteBufferPublisher() {
+            throw new UnsupportedOperationException("Streaming access not yet implemented for servlet");
+        }
+
+        @Override
+        public CompletableFuture<? extends CloseableAvailableByteBody> buffer() {
+            if (bodyIsReadAsync) {
+                throw new UnsupportedOperationException("Body is read asynchronously, cannot get contents");
+            }
+            CompletableFuture<CloseableAvailableByteBody> dest = new CompletableFuture<>();
+            CompletableFuture<? extends CloseableAvailableByteBody> result;
+            if (buffered.compareAndSet(null, dest)) {
+                try {
+                    dest.complete(new AvailableByteArrayBody(delegate.getInputStream().readAllBytes()));
```

Comment:
shouldn't this stream be closed?

Reply:
previous impl didn't do this either but yea I think so.